src/Facet/Context.hs

```diff
@@ -47,10 +47,10 @@ Context es' ! Index i' = withFrozenCallStack $ go es' i'
     | otherwise    = go es (i - 1)
   go _           _ = error $ "Facet.Context.!: index (" <> show i' <> ") out of bounds (" <> show (length es') <> ")"
 
-lookupIndex :: Alt.Alternative m => Name -> Context -> m (Index, Quantity, Either Kind Type)
+lookupIndex :: E.Has E.Empty sig m => Name -> Context -> m (Index, Quantity, Either Kind Type)
```

Comment:
All of the patterns and lookups which had returned in `Alternative` now return in `Choose`/`Empty`. Initially this was to deal with name clashes while refactoring `unify` to use `Empty`, but after that was refactored away (and also moved into a new module) I ended up keeping these using `Empty` because it enables me to be more precise about the capabilities I’m using.